spacy/matcher/phrasematcher.pyx

```diff
@@ -155,26 +156,24 @@ cdef class PhraseMatcher:
         del self._callbacks[key]
         del self._docs[key]
 
-    def add(self, key, docs, *_docs, on_match=None):
+    def add(self, key, docs, *, on_match=None):
         """Add a match-rule to the phrase-matcher. A match-rule consists of: an ID
-        key, an on_match callback, and one or more patterns.
-
-        Since spaCy v2.2.2, PhraseMatcher.add takes a list of patterns as the
-        second argument, with the on_match callback as an optional keyword
-        argument.
+        key, a list of one or more patterns, and (optionally) an on_match callback.
 
         key (str): The match ID.
         docs (list): List of `Doc` objects representing match patterns.
         on_match (callable): Callback executed on match.
-        *_docs (Doc): For backwards compatibility: list of patterns to add
-            as variable arguments. Will be ignored if a list of patterns is
-            provided as the second argument.
 
         DOCS: https://spacy.io/api/phrasematcher#add
         """
-        if docs is None or hasattr(docs, "__call__"):  # old API
-            on_match = docs
-            docs = _docs
+        # TODO we should need only one of the first two checks here
+        if docs and isinstance(docs, Doc):
+            raise ValueError(Errors.E179.format(key=key))
+        # XXX This fails in deserialization because docs are a set
+        if docs is None or not isinstance(docs, List):
+            raise ValueError(Errors.E948.format(name="PhraseMatcher", arg_type=type(docs)))
```

Comment:
The first test here was a little further down originally. The second test wasn't here, but is consistent with the other Matchers. 

I think the specificity of the first error is good, and it would be reasonable to leave both of these or either of them. I would lean towards leaving in both.

There's also the question of whether we want to require a List here. The other Matchers do. I think that's reasonable since we're going to store a list internally.